ecs-agent/daemonimages/csidriver/options_test.go

```diff
@@ -0,0 +1,66 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetServerOptions(t *testing.T) {
+	testFunc := func(t *testing.T, additionalArgs []string) (*ServerOptions, error) {
+		flagSet := flag.NewFlagSet("test-flagset", flag.ContinueOnError)
+
+		args := append([]string{
+			"/bin/csi-driver",
+		}, additionalArgs...)
+		oldArgs := os.Args
+		defer func() { os.Args = oldArgs }()
```

Comment:
Is there a way to test flag parsing without mutating `os.Args`? The mutation seems unnecessary and brittle.

Reply:
Hmmm not sure.

Reply:
We can do that by passing an `args []string` parameter to `GetServerOptions` instead of having it read from `os.Args`. I leave the decision to you as this is a nit.

Reply:
Cool, I can do this in the another PR.